### ISSUE TITLE
Releases/archives do not build as GITID is not a string

### ISSUE THREAD

**Issue by tautschnig** [arminbiere/gimsatul#7]
Building off of "rel-1.1.1.tar.gz" resulted in:
```
$ ./configure
configure: gcc -Wall -O3 -DNDEBUG
configure: generated 'makefile'
$ make
gcc -Wall -O3 -DNDEBUG -c allocate.c
gcc -Wall -O3 -DNDEBUG -c analyze.c
gcc -Wall -O3 -DNDEBUG -c assign.c
gcc -Wall -O3 -DNDEBUG -c average.c
gcc -Wall -O3 -DNDEBUG -c backtrack.c
gcc -Wall -O3 -DNDEBUG -c barrier.c
./mkconfig.sh > config.h
gcc -Wall -O3 -DNDEBUG -c build.c
build.c: In function ‘print_banner’:
build.c:13:51: error: expected ‘:’ before numeric constant
   13 |   printf ("c Version %s%s\n", VERSION, GITID ? " " GITID : "");
      |                                                   ^
      |                                                   :
make: *** [makefile:9: build.o] Error 1
```
The workaround was to manually change `#define GITID 0` to `#define GITID "0"` in config.h.

**Comment by arminbiere** [comment]
In our internal version we have this fixed some time ago:

dc205cdd build.c  (Armin Biere 2023-10-25 09:12:31 +0200 13)   printf ("c Version %s%s%s\n", VERSION, GITID ? " " : "",
dc205cdd build.c  (Armin Biere 2023-10-25 09:12:31 +0200 14)           GITID ? GITID : "");


This will go into the next release.

**Comment by arminbiere** [comment]
Fixed in new 1.1.2 release.
